File: src/main/java/com/dmarchesseault/sudoku/SudokuConstants.java
package com.dmarchesseault.sudoku;

interface SudokuConstants
{

    //todo Should ensure CHECK_SANITY <= CHECK_SANITY_BASIC <= CHECK_SANITY_DEEP
    //todo test sanity check impact on performance
    boolean CHECK_SANITY = true;
    boolean CHECK_SANITY_BASIC = true;
    boolean CHECK_SANITY_DEEP = false;
}

File: sudoku-js/src/main/js/sudoku/SudokuConstants.js
console && console.log("### SudokuConstants.js begin");

// namespace
(function(ns)
{

    //interface SudokuConstants
    let SudokuConstants = function()
    {
        throw new ns.NotInstantiableException();
    };


    //todo Should ensure CHECK_SANITY <= CHECK_SANITY_BASIC <= CHECK_SANITY_DEEP
    //todo test sanity check impact on performance
    SudokuConstants.CHECK_SANITY = true;
    SudokuConstants.CHECK_SANITY_BASIC = true;
    SudokuConstants.CHECK_SANITY_DEEP = false;

    /// exports
    ns.SudokuConstants = SudokuConstants;

}(Module("sudoku")));

console && console.log("### SudokuConstants.js end");
